pybamm/models/base_model.py

```diff
@@ -585,6 +683,17 @@ def build_coupled_variables(self):
                         )
                     )
                     try:
+                        model_var_copy = self.variables.copy()
+                        submodel_coupled_result = submodel.get_coupled_variables(
+                            self.variables
+                        )
+                        self.variables_by_submodel[submodel_name].update(
+                            {
+                                key: submodel_coupled_result[key]
+                                for key in submodel_coupled_result
+                                if key not in model_var_copy
+                            }
+                        )
                         self.variables.update(
                             submodel.get_coupled_variables(self.variables)
```

Comment:
```suggestion
                            submodel.get_coupled_variables(submodel_coupled_result)
```

Reply:
did you mean `self.variables.update(submodel_coupled_result)`??